### PR TITLE
FC-0057: add a thin proxy API btw LMS and v1

### LINKED ISSUE #2: Implement a thin proxy between the LMS and the v1 app
This proxy should (at first) forward all requests from the LMS to the v1 app.

TODO:
- [ ] Create one url to catch all API requests
- [ ] Create one view to redirect all requests to the `http://forum:{{ FORUM_PORT }}`
- [ ] Modify the `COMMENTS_SERVICE_URL` setting in edx-platform to point to `http://localhost:8000`
- [ ] Log requests made to the forum, including parameters.

### PR DESCRIPTION
**Pull request by Faraz32123**
- Add apps.py and entry points for the app to be detected by LMS
- create a thin proxy API, this is the API we'll be editing to first call the cs_comments_service urls then calling its own endpoints
close #2 

**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

### CODE REVIEW

File: forum/api/v1/views.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    """
+
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        """
+        Catches post requests and sends it to forum/cs_comments_service post urls
+        """
+        data = request.POST.dict()
+        logger.info(f"\n ForumProxyAPIView post request with data: {data}")
```
Comment (regisb) [inline forum/api/v1/views.py:26]: I don't think we should prefix log statements with new lines `\n`.

File: forum/api/v1/views.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        """
+        Catches post requests and sends it to forum/cs_comments_service post urls
+        """
+        request_url = request.get_full_path()
+        if request_url.endswith("create_comment_thread"):
+            response = ForumClient().create_comment_thread(request)
```
Comment (taimoor-ahmed-1) [inline forum/api/v1/views.py:28]: this might not scale well once we start adding more apis.

File: forum/api/v1/views.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    """
+
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        """
+        Catches post requests and sends it to forum/cs_comments_service post urls
+        """
+        request_url = request.get_full_path()
+        if request_url.endswith("create_comment_thread"):
```
Comment (regisb) [inline forum/api/v1/views.py:27]: Let's parse the url with `urllib.parse`.

File: forum/views.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+logger = logging.getLogger(__name__)
+
+
+class ForumProxyAPIView(APIView):
+    """
+    An Proxy API View to Redirect All API requests to forum/cs_comments_service urls
+    """
+
+    permission_classes = (AllowAny,)
+    COMMENTS_SERVICE_BASE_URL = "http://forum:4567"
```
Comment (taimoor-ahmed-1) [inline forum/views.py:20]: we can use `COMMENTS_SERVICE_URL`

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM

**Pull request merged**